hedera-node/src/main/java/com/hedera/services/txns/token/TokenFeeScheduleUpdateTransitionLogic.java

```diff
@@ -21,66 +21,85 @@
+import static com.hedera.services.exceptions.ValidationUtils.validateTrue;
 import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_TOKEN_ID;
-import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
-import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SUCCESS;
 import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.TOKEN_WAS_DELETED;
-
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_TOKEN_ID_IN_CUSTOM_FEES;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.TOKEN_NOT_ASSOCIATED_TO_FEE_COLLECTOR;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.CUSTOM_FEE_DENOMINATION_MUST_BE_FUNGIBLE_COMMON;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_CUSTOM_FEE_COLLECTOR;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.ROYALTY_FRACTION_CANNOT_EXCEED_ONE;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.CUSTOM_ROYALTY_FEE_ONLY_ALLOWED_FOR_NON_FUNGIBLE_UNIQUE;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.CUSTOM_FEES_LIST_TOO_LONG;
+
+/**
+ * Provides the state transition for updating token fee schedule.
+ */
 public class TokenFeeScheduleUpdateTransitionLogic implements TransitionLogic {
 	private static final Logger log = LogManager.getLogger(TokenFeeScheduleUpdateTransitionLogic.class);
-	private final TokenStore store;
+	private final TypedTokenStore typedTokenStore;
+	private final AccountStore accountStore;
 	private final TransactionContext txnCtx;
+	private final GlobalDynamicProperties globalDynamicProperties;
 
 	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
 
-	public TokenFeeScheduleUpdateTransitionLogic(final TokenStore tokenStore, final TransactionContext txnCtx) {
-		this.store = tokenStore;
+	public TokenFeeScheduleUpdateTransitionLogic(
+			final TypedTokenStore tokenStore,
+			final TransactionContext txnCtx,
+			final AccountStore accountStore,
+			final GlobalDynamicProperties globalDynamicProperties) {
+		this.typedTokenStore = tokenStore;
+		this.accountStore = accountStore;
 		this.txnCtx = txnCtx;
+		this.globalDynamicProperties = globalDynamicProperties;
 	}
 
 	@Override
 	public void doStateTransition() {
-		try {
-			transitionFor(txnCtx.accessor().getTxn().getTokenFeeScheduleUpdate());
-		} catch (Exception e) {
-			log.warn("Unhandled error while processing :: {}!", txnCtx.accessor().getSignedTxnWrapper(), e);
-			abortWith(FAIL_INVALID);
-		}
-
+		transitionFor(txnCtx.accessor().getTxn().getTokenFeeScheduleUpdate());
 	}
 
 	private void transitionFor(TokenFeeScheduleUpdateTransactionBody op) {
-		final var id = store.resolve(op.getTokenId());
-		if (id == MISSING_TOKEN) {
-			txnCtx.setStatus(INVALID_TOKEN_ID);
-			return;
+		if (!op.hasTokenId()) {
+			throw new InvalidTransactionException(INVALID_TOKEN_ID);
```

Comment:
This check is repeated - you have it in the `validate` method. It can be safely removed